tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Test_ProductOfConditions.cpp

```diff
@@ -0,0 +1,332 @@
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+    CHECK_VARIABLES_APPROX(mutable_variables, expected_mutable_variables);
+  }
+  if constexpr (bc_type ==
+                    evolution::BoundaryConditions::Type::TimeDerivative or
+                bc_type == evolution::BoundaryConditions::Type::
+                               GhostAndTimeDerivative) {
+    auto gh_result = gh_bc_helper::dg_time_derivative(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_gh_condition);
+    auto valencia_result = valencia_bc_helper::dg_time_derivative(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_valencia_condition);
+    auto product_result = product_bc_helper::dg_time_derivative(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+    if (gh_result.has_value()) {
+      if (valencia_result.has_value()) {
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+    CHECK_VARIABLES_APPROX(mutable_variables, expected_mutable_variables);
+  }
+  if constexpr (bc_type == evolution::BoundaryConditions::Type::Outflow) {
+    auto gh_result = gh_bc_helper::dg_outflow(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_gh_condition);
+    auto valencia_result = valencia_bc_helper::dg_outflow(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_valencia_condition);
+    auto product_result = product_bc_helper::dg_outflow(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+    if (gh_result.has_value()) {
+      if (valencia_result.has_value()) {
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GhValenciaDivClean.ProductOfConditions",
+    "[Unit][Evolution]") {
+  // scoped to separate out each product combination
+  {
+    INFO("Product condition of DirichletAnalytic in each system");
+
+    const grmhd::ValenciaDivClean::BoundaryConditions::DirichletAnalytic
+        valencia_condition{};
+    const GeneralizedHarmonic::BoundaryConditions::DirichletAnalytic<3_st>
+        gh_condition{};
+    TestHelpers::test_creation<std::unique_ptr<
```

Comment:
why not use the output of the creation test, serialize+deserialize, then pass the deserialized copy to the test? That might be the most rigorous :shrug: